src/huggingface_hub/fastai_utils.py

```diff
@@ -0,0 +1,393 @@
+        )
+
+    if current_fastcore_version < packaging.version.Version(fastcore_min_version):
+        raise ImportError(
+            f"`push_to_hub_fastai` and `from_pretrained_fastai` require a fastcore>={fastcore_min_version} version, but you are using fastcore version {get_fastcore_version()} which is incompatible. Upgrade with `pip install fastcore==1.3.27`."
+        )
+
+
+def check_fastai_fastcore_pyproject_versions(
+    storage_folder: str,
+    fastai_min_version: Optional[str] = "2.4",
+    fastcore_min_version: Optional[str] = "1.3.27",
+):
+    """
+    Checks that the `pyproject.toml` file in the directory `storage_folder` has fastai and fastcore versions
+    that are compatible with `save_fastai_learner`, `from_pretrained_fastai` and `push_to_hub_fastai`.
+
+    Params:
+        storage_folder (:obj:`str`):
+            Folder to look for the `pyproject.toml` file.
+        fastai_min_version (:obj:`str`, `optional`):
+            The minimum fastai version supported.
+        fastcore_min_version (:obj:`str`, `optional`):
+            The minimum fastcore version supported.
+
+    Raises:
+        :class: `FileNotFoundError`: if the there is no `pyproject.toml` in the repository that contains the fastai `Learner`.
+
+        :class: `ImportError`: if the `toml` module is not installed.
+
+        :class: `ImportError`: if the `pyproject.toml` does not indicate a version for fastai or fastcore.
+
+        :class: `ImportError`: if the `pyproject.toml` indicates a lower than minimum supported version of fastai or fastcore.
+    """
+
+    try:
+        import toml
+    except ModuleNotFoundError:
+        raise ImportError(
+            "`push_to_hub_fastai` and `from_pretrained_fastai` require the toml module. Install it with `pip install toml`."
+        )
+
+    # Check that a `pyproject.toml` exists in the repository, and if so, get a list of required packages
+    try:
+        package_versions = toml.load(f"{storage_folder}/pyproject.toml")[
+            "build-system"
+        ]["requires"]
+    except FileNotFoundError:
+        raise FileNotFoundError(
+            "There is no `pyproject.toml` in the repository that contains the fastai `Learner`. This is necessary to verify that your fastai and fastcore versions are compatible with those of the model you want to load."
+        )
+
+    # Check that `pyproject.toml` contains versions for fastai and fastcore. If there is no version available, it throws an error
+    # Then get the fastai and fastcore versions in `pyproject.toml`. If there is none, it means that it defaults to the highest version.
+    try:
+        fastai_version = str(
+            [package for package in package_versions if package.startswith("fastai")][0]
+        )
+        fastai_version = fastai_version.partition("=")[2]
```

Comment:
should we be parsing them using `packaging` and `packaging.markers` instead? It would also allow for more flexible version specifications.

Reply:
Thanks Adrin, I tried this but have some troubles, and was taking too long. Will review it soon again. Meanwhile, I registered it in issue #848.